dap/BESDapResponseBuilder.cc

```diff
@@ -284,24 +272,104 @@ int BESDapResponseBuilder::get_timeout() const
     return d_timeout;
 }
 
+/**
+ * Turn on the alarm. The code will timeout after d_timeout
+ * seconds unless timeout_off() is called first.
+ *
+ * @deprecated
+ */
+void
+BESDapResponseBuilder::timeout_on() const
+{
+#if USE_LOCAL_TIMEOUT_SCHEME
+#ifndef WIN32
+    alarm(d_timeout);
```

Comment:
On January 4, 2016 at 5:49:44 PM, Patrick West (notifications@github.com) wrote:

In dap/BESDapResponseBuilder.cc:

> @@ -284,24 +272,104 @@ int BESDapResponseBuilder::get_timeout() const
>      return d_timeout;
>  }
>   
> +/**
> + * Turn on the alarm. The code will timeout after d_timeout
> + * seconds unless timeout_off() is called first.
> + *
> + * @deprecated
> + */
> +void
> +BESDapResponseBuilder::timeout_on() const
> +{
> +#if USE_LOCAL_TIMEOUT_SCHEME
> +#ifndef WIN32
> +    alarm(d_timeout);
We have a SignalHandler class that could handle some of this, if not all of this. Wouldn't have to worry about using the macros everywhere. timeout_on would call a method on the SignalHandler instance to turn on timeout, turn off timeout. The underlying code is encapsulated in the SignalHandler class.

SignalHandler (and AlarmHandler): I assume you mean the code in libdap are not really that great. They are one of those OOP ideas that never really panned out. My opinion (I copied those from Dr Dobbs quite some time ago) is that it’s better to make the basic unix stuff follow the common patterns. By regardless of that, this is about setting the stage for removing the code above. Note that USE_LOCAL… is #undef at the top of the file.


—
Reply to this email directly or view it on GitHub.


-- 
James Gallagher
jgallagher@opendap.org